test/e2e/performanceprofile/functests/2_performance_update/memorymanager.go

```diff
@@ -0,0 +1,808 @@
+				TopologyPolicy: &policy,
+			}
+			By("Updating Performance profile")
+			profiles.UpdateWithRetry(profile)
+
+			By("Applying changes in performance profile and waiting until mcp will start updating")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdating, corev1.ConditionTrue)
+
+			By("Waiting for MCP being updated")
+			mcps.WaitForCondition(performanceMCP, machineconfigv1.MachineConfigPoolUpdated, corev1.ConditionTrue)
+		})
+
+		It("[test_id:60545] Reject guaranteed pod requesting resources that cannot be satisfied by 2 numa nodes together", Label("gu", "restricted", "rejectpod"), func() {
+			var mm1 MMPod
+			mm1.memory = "200Mi"
+			mm1.cpu = "2"
+			mm1.noOfhpgs = "24Mi"
+			mm1.hpgSize = profile.Spec.HugePages.Pages[0].Size
+			guPod := true
+			testPod := mm1.createPodTemplate(profile, guPod, &workerRTNodes[0])
+			By("creating test pod")
+			err = testclient.Client.Create(context.TODO(), testPod)
+			Expect(err).ToNot(HaveOccurred())
+			err = pods.WaitForCondition(testPod, corev1.PodConditionType(corev1.PodFailed), corev1.ConditionFalse, 2*time.Minute)
+			err := checkPodEvent(testPod, "TopologyAffinityError")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(testPod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed), "Test pod does not have QoS class of Guaranteed")
+			err = mm1.removePod(testPod)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("[test_id:60694] Accept guaranteed pod requesting resources that can be satisfied by 2 numa nodes together", Label("gu", "restricted", "allowpod"), func() {
+			var mm2 MMPod
+			mm2.hpgSize = profile.Spec.HugePages.Pages[0].Size
+			targetNode := &workerRTNodes[0]
+			mm2.memory = "200Mi"
+			mm2.cpu = fmt.Sprintf("%d", len(isolated)-2)
+			// no. of hugepages is 20 * 2 (numazones). 40Mi
+			// we are asking for 30Mi, so it needs 2 numazones combined to
+			// satisfythe requirement
+			mm2.noOfhpgs = "30Mi"
+			testPod := mm2.createPodTemplate(profile, true, targetNode)
+			initializePod(testPod, "0-1", targetNode)
+			err = mm2.removePod(testPod)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It("[test_id:60695] Allow burstable pod with hugepages", Label("burstable", "restricted", "allowpod"), func() {
+			var mm2 MMPod
+			mm2.hpgSize = profile.Spec.HugePages.Pages[0].Size
+			targetNode := &workerRTNodes[0]
+			mm2.memory = "200Mi"
+			mm2.cpu = fmt.Sprintf("%d", len(isolated)-2)
+			mm2.noOfhpgs = "8Mi"
+			testPod := mm2.createPodTemplate(profile, false, targetNode)
+			initializePod(testPod, "0-1", targetNode)
+			err := mm2.removePod(testPod)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterAll(func() {
+			By("Reverting the Profile")
+			fmt.Println("afterall")
+			profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
+			Expect(err).ToNot(HaveOccurred())
+			currentSpec, _ := json.Marshal(profile.Spec)
+			spec, _ := json.Marshal(initialProfile.Spec)
+			// revert only if the profile changes.
```

Comment:
Patching a profile without changes wouldn't return immediately anyway?

Reply:
actually it will not complain and tries to apply the patch , though no error will be returned . but it will cause a  wait time.